tests/go_test.go

```diff
@@ -448,6 +450,26 @@ func CheckMutateBuffer(org []byte, offset flatbuffers.UOffsetT, fail func(string
 	}
 }
 
+func CheckObjectAPI(buf []byte, offset flatbuffers.UOffsetT, fail func(string, ...interface{})) {
+	monster := example.GetRootAsMonster(buf, offset).UnPack()
+
+	if got := monster.Hp; 80 != got {
+		fail(FailString("hp", 80, got))
+	}
+
+	// default
+	if got := monster.Mana; 150 != got {
+		fail(FailString("mana", 150, got))
+	}
+
+	builder := flatbuffers.NewBuilder(0)
+	builder.Finish(example.MonsterPack(builder, monster))
+	monster2 := example.GetRootAsMonster(builder.FinishedBytes(), 0).UnPack()
+	if !reflect.DeepEqual(monster, monster2) {
```

Comment:
How about also calling the same code that tests a normal non-object API buffer here? That would be a less fragile test that will tell us more when it goes wrong

Reply:
bump.. can this be addressed?

Reply:
You mean test struct member by member? 
We have several references to `reflect.DeepEqual`